### ISSUE TITLE
Proposal for finding local album covers based on complex mapping from track metadata to local file paths

### ISSUE THREAD

**Issue by Oleksiy-Yakovenko** [DeaDBeeF-Player/deadbeef#1131]
Original [issue 1229](https://code.google.com/p/ddb/issues/detail?id=1229) created by Alexey-Yakovenko on 2014-11-11T13:22:05.000Z:

my issue is :
I have one folder for one artist, with several songs in it coming from several albums (and no embedded images).
I also have in the same folder the different album cover pictures, named after the different albums.

the relation between songs and covers is like:
track1 --&gt; imagealbum1.jpg
track2 --&gt; imagealbum1.jpg
track3 --&gt; imagealbum2.jpg
track4 --&gt; imagealbum2.jpg
track5 --&gt; imagealbum3.jpg

the &quot;problem&quot; is that when i play track4 or track5 for example, the &quot;album artwork&quot; plugin will show the cover imagealbum1.jpg (as there are non embedded images, i guess the plugin takes the first picture in the folder).
(the issue remains the same if tracks are flac or ogg files).

Is it possible, in the &quot;album artwork&quot; plugin, to adapt the syntax for the choice of the cover art, so that we can choose the corresponding album cover?
as in foobar for example, where you can add the syntax &quot;%album%.jpg&quot;, where %album% is the name of the album.

thanks

(running deadbeef 0.6.2 on archlinux/manjaro x86_64)

**Comment by Soukyuu** [comment]
I have something similar, but with multi disc albums. I've embedded a different cover for the two discs. Both albums have the same `%album%` value, so deadbeef only extracts the cover of the first disc and uses it for the second disc as well.

Suggestion: add a tf field in the config of the album art plugin, allowing user to set the "file name" of the generated cover cache entries.

**Comment by Alexey-Yakovenko** [comment]
agreed, this is a good idea, and very simple to do.

**Comment by KernelModule** [comment]
> Suggestion: add a tf field in the config of the album art plugin, allowing user to set the "file name" of the generated cover cache entries.
Hm, I don't get it. Please, can you explain it?

**Comment by Soukyuu** [comment]
They could use something like `imagename%tracknumber%.jpg` for matching, which would match 
track1 --> imagealbum1.jpg
track2 --> imagealbum2.jpg
track3 --> imagealbum3.jpg
track4 --> imagealbum4.jpg
track5 --> imagealbum5.jpg

It's not completely ideal, but close to it, as I don't really see how else one would manage that.

**Comment by KernelModule** [comment]
Yeah, for example it won't work if track1 and album covers is different, right?
